integration-tests/smoke/ccip/ccip_gas_price_updates_test.go

```diff
@@ -0,0 +1,125 @@
+			params.CommitOffChainConfig.RemoteGasPriceBatchWriteFrequency = *config.MustNewDuration(gasPriceExpiry)
+			return params
+		},
+	})
+	state, err := changeset.LoadOnchainState(e.Env)
+	require.NoError(t, err)
+	require.NoError(t, changeset.AddLanesForAll(e.Env, state))
+
+	allChainSelectors := maps.Keys(e.Env.Chains)
+	assert.GreaterOrEqual(t, len(allChainSelectors), 2, "test requires at least 2 chains")
+
+	sourceChain1 := allChainSelectors[0]
+	sourceChain2 := allChainSelectors[1]
+
+	feeQuoter1 := state.Chains[sourceChain1].FeeQuoter
+	feeQuoter2 := state.Chains[sourceChain2].FeeQuoter
+
+	// get initial chain fees
+	chainFee2, err := feeQuoter1.GetDestinationChainGasPrice(&bind.CallOpts{Context: ctx}, sourceChain2)
+	require.NoError(t, err)
+	chainFee1, err := feeQuoter2.GetDestinationChainGasPrice(&bind.CallOpts{Context: ctx}, sourceChain1)
+	require.NoError(t, err)
+	t.Logf("initial chain1 fee (stored in chain2): %v", chainFee1)
+	t.Logf("initial chain2 fee (stored in chain1): %v", chainFee2)
+
+	// get latest price updates sequence number from the offRamps
+	offRampChain1 := state.Chains[sourceChain1].OffRamp
+	offRampChain2 := state.Chains[sourceChain2].OffRamp
+	priceUpdatesSeqNumChain1, err := offRampChain1.GetLatestPriceSequenceNumber(&bind.CallOpts{Context: ctx})
+	require.NoError(t, err)
+	priceUpdatesSeqNumChain2, err := offRampChain2.GetLatestPriceSequenceNumber(&bind.CallOpts{Context: ctx})
+	require.NoError(t, err)
+	t.Logf("priceUpdatesSeqNumChain1: %v", priceUpdatesSeqNumChain1)
+	t.Logf("priceUpdatesSeqNumChain2: %v", priceUpdatesSeqNumChain2)
+
+	// update the price of chain2
+	tx, err := feeQuoter1.UpdatePrices(e.Env.Chains[sourceChain1].DeployerKey, fee_quoter.InternalPriceUpdates{
+		TokenPriceUpdates: nil,
+		GasPriceUpdates: []fee_quoter.InternalGasPriceUpdate{
+			{DestChainSelector: sourceChain2, UsdPerUnitGas: big.NewInt(5123)},
+		},
+	})
+	require.NoError(t, err)
+	_, err = deployment.ConfirmIfNoError(e.Env.Chains[sourceChain1], tx, err)
+	require.NoError(t, err)
+
+	// assert that the chain fees are updated by the commit plugin reports
+	priceDeviationChecked := false // flag to check if price deviation condition was met
+	assert.Eventually(t, func() bool {
+		// offRamps should have updated the sequence number
+		priceUpdatesSeqNumChain1Now, err := offRampChain1.GetLatestPriceSequenceNumber(&bind.CallOpts{Context: ctx})
+		require.NoError(t, err)
+		priceUpdatesSeqNumChain2Now, err := offRampChain2.GetLatestPriceSequenceNumber(&bind.CallOpts{Context: ctx})
+		require.NoError(t, err)
+		t.Logf("priceUpdatesSeqNumChain1: %v", priceUpdatesSeqNumChain1Now)
+		t.Logf("priceUpdatesSeqNumChain2: %v", priceUpdatesSeqNumChain2Now)
+		if priceUpdatesSeqNumChain1Now <= priceUpdatesSeqNumChain1 {
+			return false
+		}
+		if priceUpdatesSeqNumChain2Now <= priceUpdatesSeqNumChain2 {
+			return false
+		}
+
+		chainFee2Now, err := feeQuoter1.GetDestinationChainGasPrice(&bind.CallOpts{Context: ctx}, sourceChain2)
```

Comment:
Just an idea, but maybe it would be easier to understand the code if we name these as `sourceChain` and `destChain`. I think it would be easier to read it than `chain1` and `chain2`